### PR TITLE
Implements Robintrack BaseData and example algorithms

### PR DESCRIPTION
**Pull request by gsalaz98**
#### Description

* Implements BaseData for Robintrack alternative data
* Adds example algorithms

#### Related Issue

N/A

#### Motivation and Context

Alternative data efforts

#### Requires Documentation Change

N/A - new docs will be required on QC website

#### How Has This Been Tested?

Example algorithms

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [ ] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Data/Custom/Robintrack/RobintrackHoldings.cs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        /// </summary>
+        public decimal UniverseHoldingPercent => UsersHolding / TotalUniqueHoldings;
+
+        /// <summary>
+        /// Alias for <see cref="UsersHolding"/>
+        /// </summary>
+        public override decimal Value
+        {
+            get { return UsersHolding; }
+            set { UsersHolding = (int)value; }
```
Comment (Martin-Molinero) [inline Common/Data/Custom/Robintrack/RobintrackHoldings.cs:52]: Don't think these properties should have public sets?

File: Algorithm.CSharp/AltData/RobintrackHoldingsAlgorithm.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+using QuantConnect.Data.Custom.Robintrack;
+
+namespace QuantConnect.Algorithm.CSharp.AltData
+{
+    public class RobintrackHoldingsAlgorithm : QCAlgorithm
+    {
+        Symbol aapl;
+        Symbol aaplHoldings;
+        decimal lastValue;
+        bool isLong;
```
Comment (Martin-Molinero) [inline Algorithm.CSharp/AltData/RobintrackHoldingsAlgorithm.cs:26]: nit - these guys are not following our naming convention

File: Algorithm.CSharp/AltData/RobintrackHoldingsAlgorithm.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using QuantConnect.Data.Custom.Robintrack;
+
+namespace QuantConnect.Algorithm.CSharp.AltData
+{
+    public class RobintrackHoldingsAlgorithm : QCAlgorithm
```
Comment (Martin-Molinero) [inline Algorithm.CSharp/AltData/RobintrackHoldingsAlgorithm.cs:21]: I think might be nice to add some description for the algorithms

File: Common/Data/Custom/Robintrack/RobintrackHoldings.cs [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        public override bool RequiresMapping()
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Provides list of supported resolutions
+        /// </summary>
+        /// <returns>List of supported resolutions</returns>
+        public override List<Resolution> SupportedResolutions()
```
Comment (Martin-Molinero) [inline Common/Data/Custom/Robintrack/RobintrackHoldings.cs:153]: How often are these data points released? 
Maybe could keep a static list with the supported resolutions

File: Common/Data/Custom/Robintrack/RobintrackHoldings.cs [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        /// </summary>
+        /// <param name="config">Subscription config</param>
+        /// <param name="line">Line to parse</param>
+        /// <param name="date">Date that the data is being read</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>Instance of <see cref="RobintrackHoldings"/> casted as <see cref="BaseData"/></returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            var instance = Read(line);
+            instance.Symbol = config.Symbol;
```
Comment (Martin-Molinero) [inline Common/Data/Custom/Robintrack/RobintrackHoldings.cs:88]: What about live trading?

File: Common/Data/Custom/Robintrack/RobintrackHoldings.cs [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            };
+        }
+
+        /// <summary>
+        /// Parses a line of Robintrack data with an optional dateFormat
+        /// </summary>
+        /// <param name="line">Line to parse</param>
+        /// <param name="dateFormat">Date format to parse timestamp in</param>
+        /// <returns>Instance of this class</returns>
+        public static RobintrackHoldings Read(string line, string dateFormat = "yyyyMMdd HH:mm:ss")
```
Comment (Martin-Molinero) [inline Common/Data/Custom/Robintrack/RobintrackHoldings.cs:114]: Would be nice to see some simple read unit test and maybe a json round trip too?

File: Common/Data/Custom/Robintrack/RobintrackHoldings.cs (hunk 1 above)
Comment (gsalaz98) [reply Common/Data/Custom/Robintrack/RobintrackHoldings.cs:52]: Without it being `public set`, we lose the ability to initialize the object for tests without reflection.

I am in favor of making this immutable, and also eventually applying that pattern to the rest of the BaseData implementations, though it'll definitely result in breaking changes across the board.

File: Tests/Common/Data/Custom/RobintrackTests.cs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+using System;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using QuantConnect.Data.Custom.Robintrack;
+
+namespace QuantConnect.Tests.Common.Data.Custom
+{
+    [TestFixture]
+    public class RobintrackTests
```
Comment (Martin-Molinero) [inline Tests/Common/Data/Custom/RobintrackTests.cs:24]: Nice!

**Pull request merged**